status/controller.go

```diff
@@ -24,24 +27,55 @@ import (
 )
 
 type Controller[T Object] struct {
-	kubeClient         client.Client
-	eventRecorder      record.EventRecorder
-	observedConditions sync.Map // map[reconcile.Request]ConditionSet
-	terminatingObjects sync.Map // map[reconcile.Request]DeletionTimestamp
+	gvk                           schema.GroupVersionKind
+	kubeClient                    client.Client
+	eventRecorder                 record.EventRecorder
+	observedConditions            sync.Map // map[reconcile.Request]ConditionSet
+	terminatingObjects            sync.Map // map[reconcile.Request]DeletionTimestamp
+	enableObjectBasedMetrics      bool
+	ConditionDuration             pmetrics.ObservationMetric
+	ConditionCount                pmetrics.GaugeMetric
+	ConditionCurrentStatusSeconds pmetrics.GaugeMetric
+	ConditionTransitionsTotal     pmetrics.CounterMetric
+	TerminationCurrentTimeSeconds pmetrics.GaugeMetric
+	TerminationDuration           pmetrics.ObservationMetric
 }
 
-func NewController[T Object](client client.Client, eventRecorder record.EventRecorder) *Controller[T] {
+type ControllerOpts struct {
+	EnableObjectBasedMetrics bool
```

Comment:
IMO, we should the object-prefixed by default and then we should allow you to opt-in to the other metric type where we use group/kind labels to differentiate the types. That's the "deprecated" way so we should be trying to get folks off of that as quickly as we can